test/blackbox-tests/test-cases/watching/helpers.sh

```diff
@@ -22,7 +22,16 @@ with_timeout () {
 
 stop_dune () {
     with_timeout dune shutdown;
-    wait $DUNE_PID;
+    # On Linux, we may run into a bash pid aliasing bug that causes wait to
+    # reject the pid. Therefore we wait for all child processes instead.
+    if [ "$(uname -s)" = "Linux" ]
+    then
+        # wait for all child processes
+        wait $(jobs -p);
```

Comment:
Could you use phil's tail hack instead of this. it seems like it's a more faithful emulation of waiting for `$pid` than waiting for all jobs.